webpack/assets/javascripts/react_app/redux/API/APIRequest.js

```diff
@@ -1,24 +1,45 @@
 import { API } from './';
+import { actionTypeGenerator } from './APIActionTypeGenerator';
+import {
+  formatSuccess as defaultFormatSuccess,
+  formatFailure as defaultFormatFailure,
+  formatPending as defaultFormatPending,
+} from './APIHelpers';
 
-export const get = async (payload, url, store, actionTypes) => {
-  store.dispatch({
-    type: actionTypes.REQUEST,
+export const get = async (action, store) => {
```

Comment:
It would be nice to have something like:
```js
const API_REQUEST_TYPES = { GET: 'get', POST: 'post', PUT: 'put', DELETE: 'delete' };
const apiRequest = (type) => { ... };

export const get = apiRequest(API_REQUEST_TYPES.GET);
export const post = apiRequest(API_REQUEST_TYPES.POST);
export const put = apiRequest(API_REQUEST_TYPES.PUT);
export const delete = apiRequest(API_REQUEST_TYPES.DELETE);
```

Reply:
I have a PR to add all operations in #7261